hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ConcurrentSkipListMapStreamSerializer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+/**
+ * The {@link Map} serializer
+ */
+public class ConcurrentSkipListMapStreamSerializer extends AbstractMapStreamSerializer {
+    @Override
+    public int getTypeId() {
+        return SerializationConstants.JAVA_DEFAULT_TYPE_CONCURRENT_SKIP_LIST_MAP;
+    }
+
+    @Override
+    protected Map createMap(int size) {
+        return new ConcurrentSkipListMap();
```

Comment:
Minor: Maybe a special note in javadoc/ref manual would be that we don't serialise/deserialise custom comparators - `ConcurrentSkipListMap#comparator`, `ConcurrentSkipListMap(java.util.Comparator<? super K>)`, same for `ConcurrentSkipListSetStreamSerializer`, `TreeMapStreamSerializer` and `TreeSetStreamSerializer`. Hm, maybe we should even fail if a custom comparator is used? As the deserialised collection will be ordered differently.

Sorry I didn't realise this before when we were discussing the collections to support, I just realised it while reviewing.

Reply:
Will add the comparator during serialization. In addition to the comparator the following may be serialized as well: loadfactor(HashMap, ConcurrentHashMap, LinkedHashMap, HashSet, LinkedHashSet), boolean accessOrder(LinkedHashMap), boolean fair(ArrayBlockingQueue accessed using public isFair() method), boolean  fair for SynchronousQueue can not be accessed(write a Javadoc note).